### PR TITLE
Add AssessorId to LandmarkAssessment Table

### PR DESCRIPTION
**Pull request by UncleBu1728**
Add `assessorId` in `LandmarkAssessment`to allow permitted users to access specific Landmark assessment histories.

### CODE REVIEW

**Review by weihanwang — APPROVED** [review]

**Review by weihanwang — CHANGES_REQUESTED** [review]
assessorId can't be null

**Comment by UncleBu1728** [comment]
> assessorId can't be null

Does it mean we must assign a mentor to the record whenever we submit a landmark assessment?

**Comment by weihanwang** [comment]
Yes. the assessor of self assessments is the current user.

**Review by weihanwang — DISMISSED** [review]

**Comment by UncleBu1728** [comment]
`@BelongsTo` is used instead of `@HasOne` in order to associate assessor with User model using assessorId. Otherwise, an error would be shown as below:

"throw new Error(`Foreign key for "${relatedClass.name}" is missing ` +
              ^Error: Foreign key for "LandmarkAssessment" is missing on "user"."

File: src/api/database/models/User.ts [hunk 1]
```diff
@@ -121,6 +122,9 @@ class User extends Model {
   @BelongsTo(() => User, { foreignKey: 'pointOfContactId' })
   pointOfContact: User | null;
 
+  @HasMany(() => LandmarkAssessment)
+  landmarkAssessments: LandmarkAssessment[];
```
Comment (weihanwang) [inline src/api/database/models/User.ts:126]: Don't add it for now. Let's add it only when we actual need it.

File: src/api/database/models/map/LandmarkAssessment.ts [hunk 2]
```diff
@@ -23,6 +24,13 @@ class LandmarkAssessment extends Model {
   @Column(UUID)
   userId: string;
 
+  @ForeignKey(() => User)
+  @Column(UUID)
+  assessorId: string | null;
```
Comment (weihanwang) [inline src/api/database/models/map/LandmarkAssessment.ts:29]: make a comment about what null means.

File: src/api/database/models/User.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ import Interview from "./Interview";
 import GroupUser from "./GroupUser";
 import Mentorship from "./Mentorship";
 import { MenteeStatus, zMenteeStatus } from "../../../shared/MenteeStatus";
+import LandmarkAssessment from "./map/LandmarkAssessment";
```
Comment (weihanwang) [inline src/api/database/models/User.ts:29]: remove this

File: src/api/database/models/map/LandmarkAssessment.ts [hunk 4]
```diff
@@ -23,6 +24,15 @@ class LandmarkAssessment extends Model {
   @Column(UUID)
   userId: string;
 
+  // If assessorId (assessor) is null, the landmark assessment is self-evaluated by the user;
```
Comment (weihanwang) [inline src/api/database/models/map/LandmarkAssessment.ts:27]: is this line > 80 chars long?

**Pull request merged**